### DISCUSSION TITLE
Why i'm getting ECONNREFUSED ?

### DISCUSSION THREAD

**Discussion by naveen8801** [vercel/next.js#63183] [Help]
### Summary

While calling a GET endpoint of my next API server using Axios, I'm getting the following error, if I try to access this GET endpoint from the browser directly it works fine!

### Additional information

```js
{
  error: [
    Error: connect ECONNREFUSED ::1:80
        at createConnectionError (node:net:1634:14)
        at afterConnectMultiple (node:net:1664:40)
        at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
      errno: -61,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '::1',
      port: 80
    },
    Error: connect ECONNREFUSED 127.0.0.1:80
        at createConnectionError (node:net:1634:14)
        at afterConnectMultiple (node:net:1664:40)
        at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
      errno: -61,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '127.0.0.1',
      port: 80
    }
  ]
}
```
```


### Example

_No response_

**Comment by icyJoseph** [comment]
Hi,

If you are doing this during build time, it is expected, because there's no server available to handle the incoming request. This is a very typical issue. 

The fix is to reach for your data source directly, and skip jumping to an API Route, or Route Handler.

There's a dedicated page for this in the community driven Next.js FAQ:

- https://nextjs-faq.com/fetch-api-in-rsc
- https://nextjs-faq.com/fetch-api-in-getserversideprops

Please read through that, make sure you understand the issue and fix.

**Reply by naveen8801** [reply]
Thanks @icyJoseph for these links, super helpful resources ❤️
